### PR TITLE
E2E

### PR DESCRIPTION
**Pull request by cnuland**
Fixed E2E tests
Added Helm template test
Removed duplicate files for testing
Added KohlsTechnology quay images URL locations

### CODE REVIEW

**Comment by etsauer** [comment]
@cnuland does this also possibly resolve https://github.com/KohlsTechnology/eunomia/issues/8 ?

**Comment by Smiley73** [comment]
What's going on with this PR? @cnuland are you still working on it?

**Comment by cnuland** [comment]
Yes, plan to resolve it over the weekend

File: examples/seeds/simple.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  parameterSource:
+    contextDir: simple/parameters
+  triggers:
+  - type: Change
+  - type: Webhook
+    secret: ciao
+  - type: Periodic
+    cron: "0 * * * *"
+  serviceAccountRef: gitops
+  templateProcessorImage: quay.io/cnuland/kube-gitops-ocp-templates:v0.0.1
```
Comment (Smiley73) [inline examples/seeds/simple.yaml]: This should use the kohlstechnology org

File: examples/simple/helm/parameters/values.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+image:
+  repository: hello-world
+  tag: latest
+  pullPolicy: IfNotPresent
```
Comment (Smiley73) [inline examples/simple/helm/parameters/values.yaml]: Please add a empty line at the end

File: examples/seeds/simple.yaml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+apiVersion: gitops.systems/v1alpha1
+kind: GitOpsConfig
+metadata:
+  name: simple-test
+  namespace: gitops-operator
+spec:
+  # Add fields here
+  templateSource:
+    uri: https://github.com/cnuland/gitops-operator-example
```
Comment (seanmalloy) [inline examples/seeds/simple.yaml]: Please change the `cnuland` reference.

**Comment by Smiley73** [comment]
@cnuland when do you think we can wrap this one up?

File: examples/seeds/simple.yaml (hunk 3 above)
Comment (cnuland) [reply examples/seeds/simple.yaml]: done

File: examples/seeds/simple.yaml [hunk 4]
```diff
@@ -0,0 +1,23 @@
+apiVersion: gitops.systems/v1alpha1
```
[the remark this answers is not in the document]
Comment (cnuland) [reply examples/seeds/simple.yaml]: done

File: examples/simple/helm/parameters/values.yaml (hunk 2 above)
Comment (cnuland) [reply examples/simple/helm/parameters/values.yaml]: done

File: examples/seeds/simple.yaml (hunk 1 above)
Comment (cnuland) [reply examples/seeds/simple.yaml]: done

File: examples/seeds/simple.yaml [hunk 5]
```diff
@@ -2,21 +2,22 @@ apiVersion: eunomia.kohls.io/v1alpha1
...
+    contextDir: simple/parameters
   triggers:
   - type: Change
   - type: Webhook
     secret: ciao
   - type: Periodic
     cron: "0 * * * *"
   serviceAccountRef: gitops
-  templateProcessorImage: quay.io/kohlstechnology/eunomia-base:v0.0.1
+  templateProcessorImage: quay.io/kohlstechnology/gitops-ocp-templates:v0.0.1
```
Comment (seanmalloy) [inline examples/seeds/simple.yaml]: It looks like this container image does not exist. Probably should be `eunomia-ocp-templates` instead.

File: examples/seeds/simple.yaml [hunk 6]
```diff
@@ -2,21 +2,22 @@ apiVersion: eunomia.kohls.io/v1alpha1
...
   name: simple-test
+  namespace: gitops-operator
 spec:
   # Add fields here
   templateSource:
-    uri: http://localhost:8000
+    uri: https://github.com/KohlsTechnology/eunomia
     ref: master
-    contextDir: examples/templates
+    contextDir: simple/templates
```
Comment (seanmalloy) [inline examples/seeds/simple.yaml]: Looks like `simple/templates` does not exist in this repo.

File: examples/seeds/simple.yaml [hunk 7]
```diff
@@ -2,21 +2,22 @@ apiVersion: eunomia.kohls.io/v1alpha1
...
   parameterSource:
-    contextDir: examples/parameters
+    contextDir: simple/parameters
```
Comment (seanmalloy) [inline examples/seeds/simple.yaml]: Same for this one. This directory does not exist.

File: scripts/e2e-test.sh [hunk 8]
```diff
@@ -12,10 +12,17 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-oc create namespace test-gitops-operator
-oc project test-gitops-operator
-oc create configmap eunomia-templates --from-file=./templates/cronjob.yaml --from-file=./templates/job.yaml -n test-gitops-operator
-oc apply -f ./test/deploy -n test-gitops-operator
-operator-sdk test local ./test/e2e --namespace test-gitops-operator --no-setup
-oc delete project test-gitops-operator
+export JOB_TEMPLATE=$GOPATH/src/github.com/KohlsTechnology/eunomia/templates/job.yaml
```
Comment (seanmalloy) [inline scripts/e2e-test.sh]: Please add `set -e` above this line.

File: scripts/unit-tests.sh [hunk 9]
```diff
@@ -15,5 +15,5 @@
 export JOB_TEMPLATE=$GOPATH/src/github.com/KohlsTechnology/eunomia/templates/job.yaml
```
Comment (seanmalloy) [inline scripts/unit-tests.sh]: Please add `set -e` above this line.

File: test/e2e/helm_template_test.go [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "gitops-helm",
+			Namespace: namespace,
+		},
+		Spec: gitopsv1alpha1.GitOpsConfigSpec{
+			TemplateSource: gitopsv1alpha1.GitConfig{
+				URI:        "https://github.com/kohlstechnology/eunomia/examples",
+				Ref:        "master",
+				ContextDir: "simple/helm",
```
Comment (seanmalloy) [inline test/e2e/helm_template_test.go]: I think we should set `ContextDir` to `examples/simple/helm`.

File: test/e2e/helm_template_test.go [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+			Kind:       "GitOpsConfig",
+			APIVersion: "eunomia.kohls.io/v1alpha1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "gitops-helm",
+			Namespace: namespace,
+		},
+		Spec: gitopsv1alpha1.GitOpsConfigSpec{
+			TemplateSource: gitopsv1alpha1.GitConfig{
+				URI:        "https://github.com/kohlstechnology/eunomia/examples",
```
Comment (seanmalloy) [inline test/e2e/helm_template_test.go]: Change `URI` to `https://github.com/kohlstechnology/eunomia`

File: test/e2e/helm_template_test.go [hunk 12]
```diff
@@ -0,0 +1,76 @@
...
+				Ref:        "master",
+				ContextDir: "simple/helm",
+			},
+			ParameterSource: gitopsv1alpha1.GitConfig{
+				URI:        "https://github.com/kohlstechnology/eunomia/examples",
+				Ref:        "master",
+				ContextDir: "simple/helm",
```
Comment (seanmalloy) [inline test/e2e/helm_template_test.go]: Same here. Change to `example/simple/help`

File: test/e2e/helm_template_test.go [hunk 13]
```diff
@@ -0,0 +1,76 @@
...
+			Namespace: namespace,
+		},
+		Spec: gitopsv1alpha1.GitOpsConfigSpec{
+			TemplateSource: gitopsv1alpha1.GitConfig{
+				URI:        "https://github.com/kohlstechnology/eunomia/examples",
+				Ref:        "master",
+				ContextDir: "simple/helm",
+			},
+			ParameterSource: gitopsv1alpha1.GitConfig{
+				URI:        "https://github.com/kohlstechnology/eunomia/examples",
```
Comment (seanmalloy) [inline test/e2e/helm_template_test.go]: Same here ... drop the trailing `examples`

File: test/e2e/ocp_template_test.go [hunk 14]
```diff
@@ -59,29 +52,29 @@ func ocpTemplateTestDeploy(t *testing.T, f *framework.Framework, ctx *framework.
...
+			Name:      "gitops-ocp",
 			Namespace: namespace,
 		},
 		Spec: gitopsv1alpha1.GitOpsConfigSpec{
 			TemplateSource: gitopsv1alpha1.GitConfig{
-				URI:        "https://github.com/KohlsTechnology/eunomia",
+				URI:        "https://github.com/KohlsTechnology/eunomia/examples",
 				Ref:        "master",
-				ContextDir: "example/templates",
+				ContextDir: "simple/templates",
```
Comment (seanmalloy) [inline test/e2e/ocp_template_test.go]: Same ... please use `examples/simple/templates`

File: test/e2e/ocp_template_test.go [hunk 15]
```diff
@@ -59,29 +52,29 @@ func ocpTemplateTestDeploy(t *testing.T, f *framework.Framework, ctx *framework.
...
 			},
 			ParameterSource: gitopsv1alpha1.GitConfig{
-				URI:        "https://github.com/KohlsTechnology/eunomia",
+				URI:        "https://github.com/KohlsTechnology/eunomia/examples",
 				Ref:        "master",
-				ContextDir: "example/parameters",
+				ContextDir: "simple/parameters",
```
Comment (seanmalloy) [inline test/e2e/ocp_template_test.go]: Same ... please use `examples/simple/templates`

**Comment by sanbornick** [comment]
Completed by #69

**Pull request closed without merging**